host/class/uvc/usb_host_uvc_2/examples/basic_uvc_stream/main/basic_uvc_stream.c

```diff
@@ -0,0 +1,312 @@
+static SemaphoreHandle_t device_disconnected_sem;
+static QueueHandle_t rx_frames_queue[EXAMPLE_NUMBER_OF_STREAMS];
+
+bool frame_callback(const uvc_host_frame_t *frame, void *user_ctx)
+{
+    assert(frame);
+    assert(user_ctx);
+    QueueHandle_t frame_q = *((QueueHandle_t *)user_ctx);
+    // ESP_LOGI(TAG, "Frame callback! data len: %d", frame->data_len);
+    BaseType_t result = xQueueSend(frame_q, &frame, 0);
+    if (pdPASS != result) {
+        ESP_LOGW(TAG, "Queue full, losing frame"); // This should never happen. We allocated queue with the same size as EXAMPLE_FRAME_COUNT
+        return true; // We will not process this frame, return it immediately
+    }
+    return false; // We only passed the frame to Queue, so we must return false and call uvc_host_frame_return() later
+}
+
+/**
+ * @brief Device event callback
+ *
+ * Apart from handling device disconnection it doesn't do anything useful
+ *
+ * @param[in] event    Device event type and data
+ * @param[in] user_ctx Argument we passed to the device open function
+ */
+static void handle_event(const uvc_host_stream_event_data_t *event, void *user_ctx)
+{
+    switch (event->type) {
+    case UVC_HOST_TRANSFER_ERROR:
+        ESP_LOGE(TAG, "USB error has occurred, err_no = %i", event->data.error);
+        break;
+    case UVC_HOST_DEVICE_DISCONNECTED:
+        ESP_LOGI(TAG, "Device suddenly disconnected");
+        ESP_ERROR_CHECK(uvc_host_stream_close(event->data.stream_hdl));
+        xSemaphoreGive(device_disconnected_sem);
+        break;
+    default:
+        ESP_LOGW(TAG, "Unsupported event: %i", event->type);
+        break;
+    }
+}
+
+/**
+ * @brief USB Host library handling task
+ *
+ * @param arg Unused
+ */
+static void usb_lib_task(void *arg)
+{
+    while (1) {
+        // Start handling system events
+        uint32_t event_flags;
+        usb_host_lib_handle_events(portMAX_DELAY, &event_flags);
+        if (event_flags & USB_HOST_LIB_EVENT_FLAGS_NO_CLIENTS) {
+            ESP_ERROR_CHECK(usb_host_device_free_all());
+        }
+        if (event_flags & USB_HOST_LIB_EVENT_FLAGS_ALL_FREE) {
+            ESP_LOGI(TAG, "USB: All devices freed");
+            // Continue handling USB events to allow device reconnection
+        }
+    }
+}
+
+static void frame_handling_task(void *arg)
+{
+    const uvc_host_stream_config_t *stream_config = (const uvc_host_stream_config_t *)arg;
+    QueueHandle_t frame_q = *((QueueHandle_t *)(stream_config->user_ctx));
+    const int uvc_index = stream_config->usb.uvc_stream_index;
+
+    while (true) {
+        uvc_host_stream_hdl_t uvc_stream = NULL;
+
+        ESP_LOGI(TAG, "Opening UVC device 0x%04X:0x%04X-%d\n\t%dx%d@%2.1fFPS...",
+                 stream_config->usb.vid, stream_config->usb.pid, uvc_index, stream_config->vs_format.h_res, stream_config->vs_format.v_res, stream_config->vs_format.fps);
+        esp_err_t err = uvc_host_stream_open(stream_config, pdMS_TO_TICKS(5000), &uvc_stream);
```

Comment:
Yep, this is what I was talking about once we discussed the logic in the uac driver. 
Here we have timeout as an argument: `pdMS_TO_TICKS(5000)`. 
So there should be no necessity to use the next timeout, if theere is no available device (line 120): `vTaskDelay(pdMS_TO_TICKS(5000));`.

In this case, the delay between two attempts will be 5000 + 5000.

Do we really need them both here?

Reply:
But for some reason internal timeout isn't working. 
Wait a moment....

Reply:
Ah, got it. 
This two different timeouts. 

In my case the problem was during `uvc_find_streaming_intf()` not in `uvc_find_and_open_usb_device()`. 
That is why I had a device (first T/O didn't elapsed), the logic goes to the next step and returns without timeout, as interface wasn't found.

Reply:
In case `uvc_host_stream_open` returns error (not timeout), it would be called immediately, flooding the log with errors.
So I added 2nd delay. The example will be improved